### PR TITLE
Feat: Added Edit Student Course; Closes #1152

### LINKED ISSUE #1152: Edit Student Course - Front End
Missed this one!  In a student profile, beside their course, there is an edit button. This still takes them to Django Admin.

Needs to be moved to front end.  Staff view only.

e.g
/admin/courses/coursestudent/1861/change/

### PR DESCRIPTION
**Pull request by 9921622**
profile -> courses -> edit button link:
image
image

### CODE REVIEW

File: src/courses/views.py [hunk 1]
```diff
@@ -167,10 +167,32 @@ def get_form_kwargs(self):
...
+        return ctx
+
     def get_success_url(self):
         return reverse('profiles:profile_detail', args=(self.object.user.profile.id, ))
 
 
+@method_decorator(staff_member_required, name='dispatch')
+class CourseStudentUpdate(NonPublicOnlyViewMixin, UpdateView):
+    model = CourseStudent
+    form_class = CourseStudentForm
```
Comment (tylerecouture) [inline src/courses/views.py:183]: Might need to extend CourseStudentForm with a few extra fields for the staff edit form.

**Comment by 9921622** [comment]
image

File: src/courses/forms.py [hunk 2]
```diff
@@ -49,6 +49,17 @@ class Meta:
     #         # widgets = {'user': forms.HiddenInput()}
 
 
+class CourseStudentStaffForm(CourseStudentForm):
+
+    class Meta:
+        model = CourseStudent
+        fields = '__all__'
+        exclude = ['user', 'active']
```
Comment (tylerecouture) [inline src/courses/forms.py:57]: I think only one of `fields` or `exclude` are required.  exclude is essentially "all except these ones". Please verify

File: src/courses/forms.py [hunk 3]
```diff
@@ -49,6 +49,17 @@ class Meta:
...
 
+class CourseStudentStaffForm(CourseStudentForm):
+
+    class Meta:
+        model = CourseStudent
+        fields = '__all__'
+        exclude = ['user', 'active'] 
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (tylerecouture) [inline src/courses/forms.py:60]: Doing nothing? Can this method be deleted?

**Review by tylerecouture — CHANGES_REQUESTED** [review]
Looking great. Some tiny cleanup requests

File: src/courses/forms.py (hunk 2 above)
Comment (9921622) [reply src/courses/forms.py:56]: removing fields didnt change anything

**Comment by 9921622** [comment]
added requested changes

**Review by tylerecouture — APPROVED** [review]

**Pull request merged**